### PR TITLE
Add windows support.

### PR DESCRIPTION
**Pull request by aleksandrm8**
Hello! Just add some changes to use your library under Windows (Visual Studio 2015).

### CODE REVIEW

File: include/nmealib/context.h [hunk 1]
```diff
@@ -20,6 +20,10 @@
 
 #include <stddef.h>
 
+#ifdef WIN32
+#define __attribute__(A) /* do nothing */
```
Comment (fhuberts) [inline include/nmealib/context.h]: I'd rather that you put this in an include file, like windowssupport.h. unconditionally include it, and in that include file use #ifdef WIN32 (how about win 64?)

File: include/nmealib/info.h [hunk 2]
```diff
@@ -21,7 +21,12 @@
 #include <nmealib/util.h>
 #include <stdbool.h>
 #include <stdint.h>
+#ifndef WIN32
 #include <sys/time.h>
+#else
+#include <Windows.h>
+#define __attribute__(A) /* do nothing */
```
Comment (fhuberts) [inline include/nmealib/info.h]: same for these 2 lines

File: include/nmealib/util.h [hunk 3]
```diff
@@ -27,7 +27,11 @@ extern "C" {
 #endif /* __cplusplus */
 
 #ifndef INLINE
-#define INLINE inline __attribute__((always_inline))
+# ifndef WIN32
+#  define INLINE inline __attribute__((always_inline))
+# else
+#  define INLINE inline
```
Comment (fhuberts) [inline include/nmealib/util.h]: just use the new include file

File: include/nmealib/util.h [hunk 4]
```diff
@@ -157,7 +161,12 @@ int nmeaAppendChecksum(char *s, size_t sz, size_t len);
  * @param format The string format to use
  * @return The number of printed characters, -1 on error
  */
+#ifndef WIN32
 int nmeaPrintf(char *s, size_t sz, const char *format, ...) __attribute__ ((format(printf, 3, 4)));
+#else
+int nmeaPrintf(char *s, size_t sz, const char *format, ...);
```
Comment (fhuberts) [inline include/nmealib/util.h]: just use the new include file

File: src/generator.c [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/*
+/*
```
Comment (fhuberts) [inline src/generator.c]: whitespace damage, remove

File: src/info.c [hunk 6]
```diff
@@ -179,12 +179,28 @@ void nmeaTimeSet(NmeaTime *utc, uint32_t *present, struct timeval *timeval) {
   }
 
   if (timeval) {
+#ifdef WIN32
+    gmtime(&timeval->tv_sec, &tm);
```
Comment (fhuberts) [inline src/info.c]: how about putting #define gmtime_r(x,y) gmtime(x,y) in the new include file?

File: src/util.c [hunk 7]
```diff
@@ -28,13 +28,20 @@
...
+# include <unistd.h>
+#endif
+
 
 /** The maximum size of a string-to-number conversion buffer*/
 #define NMEALIB_CONVSTR_BUF    64
 
 void nmeaRandomInit(void) {
+#ifdef WIN32
+  srand((unsigned int) time(NULL));
```
Comment (fhuberts) [inline src/util.c]: how about putting #define srandom(x) srand(x) in the new include file?

File: src/util.c [hunk 8]
```diff
@@ -51,7 +58,11 @@ double nmeaRandom(const double min, const double max) {
   if ((randomFile == -1) //
       || (read(randomFile, &value, sizeof(value)) != sizeof(value))) {
     /* can't be covered in a test */
+#ifdef WIN32
+    value = rand();
```
Comment (fhuberts) [inline src/util.c]: how about putting #define random(x) rand(x) in the new include file?

File: test/generator.c [hunk 9]
```diff
@@ -147,7 +147,11 @@ static void test_nmeaGeneratorInvokeStatic(void) {
 
   gettimeofday(&tp, NULL);
   r = nmeaGeneratorInvokeStatic(NULL, &info);
+#ifdef WIN32
+  gmtime_s(&tt, &tp.tv_sec);
```
Comment (fhuberts) [inline test/generator.c]: why use gmtime_s here and gmtime in on of the other files? inconsistent

File: test/generator.c [hunk 10]
```diff
@@ -299,7 +303,11 @@ static void test_nmeaGeneratorInvokeRotate(void) {
 
   gettimeofday(&tp, NULL);
   r = nmeaGeneratorInvokeRotate(NULL, &info);
+#ifdef WIN32
+  gmtime_s(&tt, &tp.tv_sec);
```
Comment (fhuberts) [inline test/generator.c]: same here

File: test/generator.c [hunk 11]
```diff
@@ -339,7 +347,11 @@ static void test_nmeaGeneratorInvokeRotate(void) {
...
```
Comment (fhuberts) [inline test/generator.c]: same here

File: test/generator.c [hunk 12]
```diff
@@ -385,7 +397,11 @@ static void test_nmeaGeneratorInvokeRotate(void) {
...
```
Comment (fhuberts) [inline test/generator.c]: same here

File: test/generator.c [hunk 13]
```diff
@@ -733,7 +749,11 @@ static void test_nmeaGeneratorInit(void) {
 
   gettimeofday(&tp, NULL);
   r = nmeaGeneratorInit(&gen, &info);
+#ifdef WIN32
+  gmtime_s(&tt, &tp.tv_sec);
```
Comment (fhuberts) [inline test/generator.c]: same here

File: test/generator.c [hunk 14]
```diff
@@ -771,7 +791,11 @@ static void test_nmeaGeneratorInit(void) {
 
   gettimeofday(&tp, NULL);
   r = nmeaGeneratorInit(g, &info);
+#ifdef WIN32
+  gmtime_s(&tt, &tp.tv_sec);
```
Comment (fhuberts) [inline test/generator.c]: same here

File: test/info.c [hunk 15]
```diff
@@ -549,7 +549,11 @@ static void test_nmeaTimeSet(void) {
   memcpy(&utcExpected, &utcClean, sizeof(utcClean));
   gettimeofday(&timeval, NULL);
   nmeaTimeSet(&utc, NULL, NULL);
+#ifdef WIN32
+  gmtime_s(&tt, &timeval.tv_sec);
```
Comment (fhuberts) [inline test/info.c]: same here

File: test/info.c [hunk 16]
```diff
@@ -565,7 +569,11 @@ static void test_nmeaTimeSet(void) {
   memcpy(&utcExpected, &utcClean, sizeof(utcClean));
   gettimeofday(&timeval, NULL);
   nmeaTimeSet(&utc, &present, NULL);
+#ifdef WIN32
+  gmtime_s(&tt, &timeval.tv_sec);
```
Comment (fhuberts) [inline test/info.c]: same here

**Comment by aleksandrm8** [comment]
@fhuberts , looks like your change requests are fixed?

**Pull request merged**